src/api/app/services/project_configuration_service/project_configuration_updater.rb

```diff
@@ -0,0 +1,22 @@
+module ProjectConfigurationService
+  class ProjectConfigurationUpdater
+    def initialize(project, user, params)
+      @user = user
+      @project = project
+      @params = params
+    end
+
+    def call
+      @params[:user] = @user.login
+      @project.config.save(sliced_params, @params[:config])
+    end
+
+    private
+
+    def sliced_params
+      sliced_params = @params.slice(:user, :comment)
+      sliced_params.permit!
+      sliced_params.to_h
```

Comment:
Why is the ``to_h`` necessary?

Reply:
since rails 5.1, ActionController::Parameters no longer inherits from Hash. Therefore you have to call the ```.to_h``` if the other method is expecting  a hash.

Reply:
```
irb(main):002:0> def foo(params: {})
irb(main):003:1>   puts params.inspect
irb(main):004:1> end
=> :foo
irb(main):005:0> foo(ActionController::Parameters.new(foo:'bar'))
Traceback (most recent call last):
        1: from (irb):5
ActionController::UnfilteredParameters (unable to convert unpermitted parameters to hash)
irb(main):006:0> 
```

Reply:
@chrisBR https://github.com/rails/rails/pull/28734 as reference

Reply:
Yes because it is not permitted

```
foo(params: ActionController::Parameters.new(foo:'bar').permit!)
# => <ActionController::Parameters {"foo"=>"bar"} permitted: true>
```

Reply:
please review it again @ChrisBr